controller/k8s/clientset.go

```diff
@@ -9,6 +10,32 @@ import (
 )
 
 func NewClientSet(kubeConfig string) (*kubernetes.Clientset, error) {
+	config, err := parseConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return clientset, nil
```

Comment:
There's no need for the nil check now -- you can just return the result of NewForConfig directly, since it's return signature is the same as this func.

```
	return kubernetes.NewForConfig(config)
```